dcicutils/deployment_utils.py

```diff
@@ -58,7 +64,282 @@ def boolean_setting(settings, key, default=None):
+        })
+
+        # make additional updates as needed
+        # XXX: Make worker tier perhaps?
+        for idx, option in enumerate(configuration):
+
+            # make it a big machine (or not if we say so)
+            if option['OptionName'] == 'InstanceType':
+                option['Value'] = cls.DEFAULT_INDEXER_SIZE if not size else size
+
+            # add the additional env variable here as well
+            if option['OptionName'] == 'EnvironmentVariables':
+                option['Value'] += ',ENCODED_INDEX_SERVER=True'
+
+        # filter '' option settings and known 'bad' options
+        # XXX: Refactor into the above loop? Don't think it really matters since this code doesn't need
+        # to be high performance and it's convenient to organize logic like this.
+        configuration = list(filter(lambda d: (d.get('Value', '') != '' and d['OptionName'] not in ['AppSource'],
+                                    configuration)))
+
+        # upload the template
+        if is_cgap_env(env_name):
+            return eb_client.create_configuration_template(
+                ApplicationName=cls.EB_APPLICATION,
+                TemplateName=CGAP_ENV_INDEXER,
+                OptionSettings=configuration,
+                EnvironmentId=cls.extract_environment_id(env_name)
+            )
+        else:
+            return eb_client.create_configuration_template(
+                ApplicationName=cls.EB_APPLICATION,
+                TemplateName=FF_ENV_INDEXER,
+                OptionSettings=configuration,
+                EnvironmentId=cls.extract_environment_id(env_name)
+            )
+
+    @classmethod
+    def create_indexer_environment(cls, env_name, app_version):
+        """ Creates a new environment for indexing based on the given env_name. Will look for a template
+            called FF_ENV_INDEXER or CGAP_ENV_INDEXER, if that does not exist this call will fail.
+
+        :param env_name: env to base template off of
+        :param app_version: application version to deploy, MUST match that running on env_name
+        :return: result of EB env creation
+        :raises RuntimeError if bad env given
+        """
+        eb_client = boto3.client('elasticbeanstalk')
+        if is_cgap_env(env_name):
+            return eb_client.create_environment(
+                ApplicationName=cls.EB_APPLICATION,
+                EnvironmentName=CGAP_ENV_INDEXER,
+                TemplateName=CGAP_ENV_INDEXER,
+                VersionLabel=app_version
+            )[cls.STATUS] == cls.UPDATING and cls.delete_indexer_template(eb_client, FF_ENV_INDEXER)
+        elif is_fourfront_env(env_name):
+            return eb_client.create_environment(
+                ApplicationName=cls.EB_APPLICATION,
+                EnvironmentName=FF_ENV_INDEXER,
+                TemplateName=FF_ENV_INDEXER,
+                VersionLabel=app_version
+            )[cls.STATUS] == cls.UPDATING and cls.delete_indexer_template(eb_client, FF_ENV_INDEXER)
```

Comment:
As noted above, this would be prettier and less confusing if `cls.` were not interfering.

Reply:
Really if there were an `indexer_name_for_env` to get the client name. These clauses seem otherwise the same and could be collapsed as they do the same thing.